tests/monotouch-test/ObjCRuntime/RegistrarTest.cs

```diff
@@ -2145,6 +2145,55 @@ public void TestCtors ()
 			}
 		}
 
+#if __MACOS__
+		[Test]
+		public void CustomUserTypeWithDynamicallyLoadedAssembly ()
+		{
+			var customTypeAssemblyPath = global::System.IO.Path.Combine (global::Xamarin.Tests.Configuration.RootPath, "tests", "test-libraries", "custom-type-assembly", ".libs", "macos", "custom-type-assembly.dll");
+			Assert.That (customTypeAssemblyPath, Does.Exist, "existence");
+
+			var size = 10;
+			var handles = new GCHandle [size];
+			var array = new NSMutableArray ();
+
+			// Create a bunch instances of a custom object the static registrar didn't know about at build time.
+			// We do this on a different thread to prevent the GC from finding these instances on the main thread's stack.
+			var thread = new Thread (() => {
+				var customTypeAssembly = global::System.Reflection.Assembly.LoadFrom (customTypeAssemblyPath);
+				var customType = customTypeAssembly.GetType ("MyCustomType");
+				for (var i = 0; i < size; i++) {
+					var obj = (NSObject) global::System.Activator.CreateInstance (customType);
+					array.Add (obj);
+					handles [i] = GCHandle.Alloc (obj, GCHandleType.Weak);
+				}
+				// Run the GC a couple of times.
+				GC.Collect ();
```

Comment:
Just wondering why you run this twice?

Reply:
In certain scenarios multiple GC collections have to occur for something to be collected.

And this test is testing that something _isn't_ collected.

So just run the GC more than once to be absolutely sure that the test isn't unintentionally passing because we're not running the GC enough times to run into failure scenarios.